_posts/2018-01-20-constructor-functions-and-function-context-in-javascript.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+date: 2018-01-20
+title: Constructor functions and function context in JavaScript
+---
+
+Today, I spent some time looking into JavaScript constructor functions. I ended up reading (again...) about function context and it finally sunk in. Here are my findings.
+
+
+
+#### Constructor Function
+
+First of all, what is a constructor function? A constructor function is a function that allows you to create an instance of an object when it is called. So every time you call it, you get a new object. It looks like this:
+
+```javascript
+function Person(name, age) {
+    this.name = name;
+    this.age = age;
+}
+```
+
+And you call it like with the `new` keyword:
```

Comment:
Sorry, missed this the first time - don't need the 'like' here 'call it with the new keyword'